server/utils/auth/github_auth_router.py

```
@@ -0,0 +1,443 @@
+        f"No GitHub credential available for user={user_id} "
+        f"repo={repo_full_name}"
+    )
+
+
+def _backfill_repo_installation(
+    user_id: str, repo_full_name: str, installation_id: int
+) -> None:
+    """Best-effort write of ``installation_id`` onto a repo row after fallback resolution."""
+    try:
+        with db_pool.get_admin_connection() as conn:
+            with conn.cursor() as cur:
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)